driver/src/main/scala/za/co/absa/hyperdrive/driver/IngestionDriver.scala

```diff
@@ -61,11 +66,19 @@ private[driver] class IngestionDriver {
 
   private def getStreamWriter(conf: Configuration): StreamWriter = StreamWriterAbstractFactory.build(conf)
 
+  private def getIngestionFinalizer(conf: Configuration): IngestionFinalizer = IngestionFinalizerAbstractFactory.build(conf)
+
   private def printConfiguration(configuration: Configuration): Unit = {
     import scala.collection.JavaConverters._
     configuration
       .getKeys
       .asScala
       .foreach(key => logger.info(s"\t$key = ${configuration.getProperty(key)}"))
   }
+
+  private def applyDefaults(configuration: Configuration): Unit = {
```

Comment:
Not sure about this bit. Wouldn't it better to allow the job to fail and then check what is wrong? One possible issue could be a functional bug, e.g. incorrect columns ingested because the user forgot to provide the desired ones but the system assumed it to be all.

Reply:
I introduced this method to avoid adding `component.finalizer=...NoOpFinalizer` to all existing configurations. I think this is a sensible preset.

However for the column selector stream transformer you are probably right, selecting all columns by default might not be as expected.

Reply:
Fair.